engines/cine/metaengine.cpp

```diff
@@ -245,6 +252,161 @@ SaveStateDescriptor CineMetaEngine::querySaveMetaInfos(const char *target, int s
 	return SaveStateDescriptor();
 }
 
+Common::KeymapArray CineMetaEngine::initKeymaps(const char *target) const {
+	using namespace Common;
+	using namespace Cine;
+
+	Keymap *engineKeyMap = new Keymap(Keymap::kKeymapTypeGame, "cine-main", "CINE main");
+	Keymap *gameKeyMap = new Keymap(Keymap::kKeymapTypeGame, "game-shortcuts", _("Game Keymappings"));
+	Keymap *mouseKeyMap = new Keymap(Keymap::kKeymapTypeGame, "mouse-shortcuts", _("Key to Mouse Keymappings"));
+
+	Action *act;
+
+	act = new Action(kStandardActionLeftClick, _("Left Click"));
+	act->setLeftClickEvent();
+	act->addDefaultInputMapping("MOUSE_LEFT");
+	act->addDefaultInputMapping("JOY_A");
+	engineKeyMap->addAction(act);
+
+	act = new Action(kStandardActionRightClick, _("Right Click"));
+	act->setRightClickEvent();
+	act->addDefaultInputMapping("MOUSE_RIGHT");
+	act->addDefaultInputMapping("JOY_B");
+	engineKeyMap->addAction(act);
+
+	act = new Action("MOUSELEFT", _("Mouse Left Keys"));
+	act->setCustomEngineActionEvent(kActionMouseLeft);
+	act->addDefaultInputMapping("RETURN");
+	act->addDefaultInputMapping("KP_ENTER");
+	act->addDefaultInputMapping("KP5");
+	mouseKeyMap->addAction(act);
```

Comment:
This seems strange, but I understand it as the handling for this checks allowPlayerInput.
I'd like to look through more of the checks of mouse handling to more clearly understand the use.
Could this possibly indicate a better named behavior that mouse button also perform, such as  of text advancing / skipping?

Reply:
I have come up with better named behavior, they are very generic

Reply:
Yes, the updated descriptions helps make sense.